apps/breethe/lib/breethe/sources/google/geocoding.ex

```diff
@@ -27,6 +32,46 @@ defmodule Breethe.Sources.Google.Geocoding do
     |> strip()
   end
 
+  def find_location_country_code(search_term) do
+    query =
+      URI.encode_query(%{
+        "address" => search_term,
+        "key" => Application.get_env(:breethe, :google_maps_api_key)
+      })
+
+    query
+    |> query_google_api()
+    |> Jason.decode!()
+    |> (& &1["results"]).()
+    |> List.first()
+    |> (& &1["address_components"]).()
+    |> find_country_code()
+  end
+
+  def find_location_country_code(lat, lon) do
+    query =
+      URI.encode_query(%{
+        "latlng" => "#{lat},#{lon}",
+        "key" => Application.get_env(:breethe, :google_maps_api_key)
+      })
+
+    query
+    |> query_google_api()
+    |> Jason.decode!()
+    |> (& &1["results"]).()
+    |> List.first()
+    |> (& &1["address_components"]).()
+    |> find_country_code()
+  end
```

Comment:
I guess there is potential for refactoring these methods and extracting a shared helper method as both are almost identical?

Reply:
Yep, I wanted to include this in the `lat, lon` refactor.

Reply:
Sure, I can do it here but it involves the whole code base which is why I thought it best to split it up?

Reply:
Ah right - yes I'd misunderstood, sorry about that!
fixed